baseapp/abci.go

```diff
@@ -624,11 +624,29 @@ func (app *BaseApp) ApplySnapshotChunk(req abci.RequestApplySnapshotChunk) abci.
+
+func queryListContainsReq(req abci.RequestQuery) bool {
+	for _, query := range stateMachineQueryList {
+		if req.Path == query {
+			return true
+		}
+	}
+	return false
+}
+
 func (app *BaseApp) handleQueryGRPC(handler GRPCQueryHandler, req abci.RequestQuery) abci.ResponseQuery {
 	ctx, err := app.CreateQueryContext(req.Height, req.Prove)
 	if err != nil {
 		return sdkerrors.QueryResultWithDebug(err, app.trace)
 	}
+	// use infinite gas metering for potential state machine queries
+	// delete in next major release
+	if queryListContainsReq(req) {
+		ctx = ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
```

Comment:
asking to learn: what does this change indicate? Why do we use infinite gas meter?